improver/cli/__init__.py

```diff
@@ -239,27 +243,43 @@ def constrained_inputcubelist_converter(to_convert):
                 than one match is found.
         """
         from improver.utilities.load import load_cube
-        from iris.cube import CubeList
-
-        cubelist = CubeList()
-        for constr in constraints:
-            constr_list = [constr] if isinstance(constr, str) else constr
-            found_cubes = []
-            for constr_item in constr_list:
-                try:
-                    found_cubes.append(maybe_coerce_with(
-                        load_cube, to_convert, constraints=constr_item))
-                except ValueError:
-                    pass
-            if len(found_cubes) != 1:
-                msg = (f"Incorrect number of valid inputs available for the "
-                       "{constr} constraint. "
-                       f"Number of valid inputs: {len(found_cubes)} "
-                       f"The valid inputs found are: {found_cubes}")
-                raise ValueError(msg)
-            cubelist.extend(found_cubes)
-
-        return cubelist
+        from iris import load_raw
+
+        if isinstance(to_convert, str):
+            cubelist = load_raw(to_convert)
+            loop = lambda x: [load_cube(
+                to_convert, constraints=j)
+                for j in x]
+        else:
+            cubelist = to_convert
+            # Constraint needs 'name=' prefix.
+            # Returns a list so selected 0 index required.
+            loop = lambda x: [extract_subcube(
+                to_convert, ["name=" + j])[0]
+                for j in x]
```

Comment:
Select if the cubes should be loaded with constraint or extracted with constraint.